packages/explat-client-react-helpers/src/index.tsx

```diff
@@ -38,32 +55,41 @@ interface ExPlatClientReactHelpers {
 			experimentAssignment: ExperimentAssignment | null
 		) => JSX.Element;
 		name: string;
+		options?: Partial< UseExperimentOptions >;
 	} ) => JSX.Element;
 }
 
 export default function createExPlatClientReactHelpers(
 	exPlatClient: ExPlatClient
 ): ExPlatClientReactHelpers {
-	const useExperiment = ( experimentName: string ): [ boolean, ExperimentAssignment | null ] => {
+	const useExperiment = (
+		experimentName: string,
+		providedOptions: Partial< UseExperimentOptions > = {}
+	): [ boolean, ExperimentAssignment | null ] => {
+		const options: UseExperimentOptions = { ...defaultUseExperimentOptions, ...providedOptions };
```

Comment:
After `ExperimentOptions.isEligible` becomes optional, this assignment can be converted to:
```js
const { isEligible = true } = providedOptions;
```

Reply:
Yeah I think I still prefer the partial pattern as we get the stricter typing on `ExperimentOptions`.

Reply:
The strict `ExperimentOption`, though, is useful only for one local internal variable. To get a nice type for it, the much more important public API types get a complex type. I believe that's exactly the wrong tradeoff to make.

That's why I'd still strongly recommend making the `ExperimentOptions` properties optional, telling the library user in the most straightforward way what a valid call looks like. And you can use `Required< Options >`, the opposite of `Partial`, for typing internal variables if needed.

There's a lot of prior art for the optional-properties-in-options recommendation. There are TypeScript's types for DOM `addEventListener` method options:

https://github.com/microsoft/TypeScript/blob/master/lib/lib.dom.d.ts#L33-L36

webpack defines plenty of `*Options` types for config values and plugins, e.g.:

https://github.com/webpack/webpack/blob/master/types.d.ts#L4900